### PR TITLE
Exclude info maps when adding elements to OM pool with GNU arg -i set (reopened)

### LINKED ISSUE #1535: Exclude info maps when adding elements to OM pool with GNU arg -i set
To save some memory at runtime, we can skip storing info maps by setting the gnu -i flag, but the current implementation in the OM still saves those data during the simulation and only drops them at the end if not needed.

- [x] Set the exclude info maps flag at the start of the simulation so that all the calls to `_add_to_pool()` in the OM will ignore the info maps if the flag was set to `True`.
- [x] Check the reloading feature still working as expected.

### PR DESCRIPTION
**Pull request by ltp23**
- [x] Added an entry to the changelog.

### What & Why
To save some memory at runtime, we can skip storing info maps by setting the gnu `-i` flag, but the current implementation in the OM still saves those data during the simulation and only drops them at the end if not needed. 

### How
- [x] Set the exclude info maps flag at the start of the simulation so that all the calls to `_add_to_pool()` in the OM will ignore the info maps if the flag was set to `True`.

#### On main branch
on-main-branch

Regardless of whether the excluding info maps flag (`-i`) is used or not, the memory usage is the same. This indicates the existing inefficiency of the OM holding onto the info maps data till the end even when they are not needed.

#### On the current branch
blue-with-info-maps---black-without-info-maps

Here you can see the difference. When the `-i` flag is used, no info maps data are stored  (black line), and when it is not used, real data will be stored along with info maps data.

### Test plan

The point of this PR is to improve memory usage internally, so the existing functionalities regarding the outputs should remain the same.

Let say I have a json filter file that has a catch-all regex filter
```
.*
```

To run the program and exclude the info maps data,
```
python main.py -i
```

### CODE REVIEW

**Review by tomhuhh — APPROVED** [review]
Thank you Loi! I think overall this is a very neat implementation. 

I didn't know how to test this most efficiently. So I decided to increase the number of cows to 5000 in the default file and run a simulation using the command `python main.py -i `. Normally it would take me 40 to 60 minutes to run a herd of this size on my computer but it only took me about 20 minutes on this branch so I'm quite happy about the outcome. 

Thanks, and please let me know if there is a better way to test this improvement in model performance or if you need anything else from me.

File: main.py [hunk 1]
```diff
@@ -129,6 +129,7 @@ def run_rufas(
     """
 
     output_manager = OutputManager()
+    output_manager.set_exclude_info_maps_flag(exclude_info_maps)
```
Comment (JoeWaddell) [inline main.py:132]: Now that this method is in place, does `run_load_vars_pool` actually need the `exclude_info_maps` input arg, or could it just check it's own (e.g. `OutputManager` flag that's been set?)

On further review, I see that this question applies to quite a few methods here.

Comment (ltp23) [reply main.py:132]: Hi Joe, there are several functions in `main.py` that have the `exclude_info_maps` parameter. I did make a PR to remove those (#1496), but to minimize the changes in `main.py`, I put it off at the moment.

**Review by jadamchick — APPROVED** [review]
Thanks Loi! I ran this with one of my main reports and compared outputs with and without the -i flag, and they were identical. Looks good to me!

**Pull request merged**